SignalShareExtension/ShareViewController.swift

```diff
@@ -664,10 +673,75 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
         // FIXME: For now, we prefer the URL provider and discard the text provider, since it's more useful to share the URL than the caption
         // but we *should* include both. This will be a bigger change though since our share extension is currently heavily predicated
         // on one itemProvider per share.
-        guard let itemProvider: NSItemProvider = type(of: self).preferredItemProvider(inputItem: inputItem) else {
-            let error = ShareViewControllerError.assertionError(description: "No item provider in input item attachments")
+
+        // Prefer a URL provider if available
```

Comment:
So a share extension has a few scenarios:

1. One shared "thing", which has one representation
2. One shared "thing", which has multiple representations (e.g. firefox URLs which also include a title/caption, etc.).
3. N shared "things" (which are all visual media).
4. N shared "things" (none or not all are visual media).

Note that in all of these cases, multiple "things" correspond to multiple attachments (`NSItemProvider`), not multiple items (`NSExtensionItem`).  

In case 3 (albums), we want N attachments (`NSItemProvider`), one for each item in the album.  It doesn't make sense to "prefer/pick" among them.  We _do_ need to pick in case 2, which we still do.